autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/coalescer/NoCoalescer.py

```diff
@@ -0,0 +1,51 @@
+from typing import Any, Dict, Optional, Union
+
+import numpy as np
+
+from autoPyTorch.pipeline.components.preprocessing.tabular_preprocessing.coalescer.base_coalescer import BaseCoalescer
+
+
+class NoCoalescer(BaseCoalescer):
+    """
+    Don't perform NoCoalescer on categorical features
+    """
+    def __init__(self,
+                 random_state: Optional[Union[np.random.RandomState, int]] = None
+                 ):
+        super().__init__()
+        self.random_state = random_state
+
+    def fit(self, X: Dict[str, Any], y: Any = None) -> BaseCoalescer:
+        """
+        The fit function calls the fit function of the underlying model
+        and returns the transformed array.
+        Args:
+            X (np.ndarray): input features
+            y (Optional[np.ndarray]): input labels
+
+        Returns:
+            instance of self
+        """
+        self.check_requirements(X, y)
+
+        return self
+
+    def transform(self, X: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Adds the self into the 'X' dictionary and returns it.
+        Args:
+            X (Dict[str, Any]): 'X' dictionary
```

Comment:
What is `X` dictionary?
`fit_dictionary`? (Ravin says the `fit_dictionary` will deprecate soon, but do you have any idea when?)

Reply:
Scikit-learn supports passing a dictionary alongside the data. See [here](https://github.com/scikit-learn/scikit-learn/blob/2beed55847ee70d363bdbfe14ee4401438fba057/sklearn/pipeline.py#L266)

It makes a lot of sense to use it instead of X as a fit_dictionary.

From all of the refactoring changes, this is to me the most important.

When** depends on when there is a contributor that wants to do this change :)

Reply:
Sorry I did not get you, so please add your ideas to the doc-string as well?
Especially, I do not get why the meaning behind 'X' in the sentence.
But still it is a bit confusing for me.
Do you know why sklearn uses `X` for both `fit_dictionary` and `feature_data`?